src/main/webapp/search-meal-script.js

```diff
@@ -17,15 +17,20 @@ function searchMeal() {
     {
         const container = document.getElementById("dishes-container");
         container.innerText = "";
-        dishes = dishes ?? [""];
-        dishes.forEach((dish) => {
-            container.appendChild(createListElement(dish));
+        dishes = dishes ?? {0: ""};
+        Object.entries(dishes).forEach((dish) => {
+            container.appendChild(createListElement(dish[1]));
         });
     });
 }
 
-function createListElement(val) {
+function createListElement(obj) {
     const listElement = document.createElement('li');
-    listElement.innerText = val;
+    const id = obj.id;
+    listElement.innerText = obj.title;
+    listElement.onclick = function() {
+        location.href = 'meal.html?id=' + id.toString();
```

Comment:
There's a special syntax for building strings in JS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

```
`meal.html?id=${id}`
```